core/src/main/java/com/datastax/oss/driver/internal/core/channel/ConnectInitHandler.java

```diff
@@ -59,8 +58,10 @@ public void connect(
     realConnectPromise.addListener(future -> onRealConnect(ctx));
 
     // Make the caller's promise wait on the other two:
-    PromiseCombiner combiner = new PromiseCombiner(ImmediateEventExecutor.INSTANCE);
-    combiner.addAll(new Future[] {realConnectPromise, initPromise});
+    // JAVA-2569: preserve binary compatibility with Netty < 4.1.34
+    @SuppressWarnings("deprecation")
+    PromiseCombiner combiner = new PromiseCombiner();
```

Comment:
I understand that `PromiseComibner()` constructor is deprecated, maybe we could extract this as a method and do deprecation inside of it? Then we will not need that duplication(https://github.com/datastax/java-driver/pull/1371/files#diff-08569c129701967a21d9a64ee733151eR179), and if someone will need to construct this class, he could call this method that encapsulates the logic?

Reply:
You mean, create a utility class exposing a static method `PromiseCombiner newPromiseCombiner()`? I think I disagree with you for two reasons: 1) the creation of that utility class requires the same amount of lines of code than the current situation so it doesn't bring concision; and 2) I bet we would forget to use this utility class and keep instantiating `PromiseCombiner` by simply doing `new PromiseCombiner()`...